### PR TITLE
finished mvp, didnt commit a lot got to sidetracked with testing

### PR DESCRIPTION
**Pull request by PatrickTheCodeGuy**
Changed some edge cases:

on string funtions, added the trim method, cause strings can have spaces on accident, trimming out the white space helps prevent that error

### CODE REVIEW

File: basic-javascript/project-1.spec.js [hunk 1]
```diff
@@ -1,3 +1,124 @@
...
+	const actual = helpers.roundNumber(3.3);
+	expect(actual).toBe(3);
+});
+test("should round a number up", () => {
+	const actual = helpers.roundUp(3.3);
+	expect(actual).toBe(4);
+});
+test("should add an exclamation point", () => {
+	expect(helpers.addExclamationPoint("  I like pie  ")).toBe("I like pie!");
+	expect(helpers.addExclamationPoint("I like pie")).toBe("I like pie!");
```
Comment (bradencodes) [inline basic-javascript/project-1.spec.js:99]: Interesting test to show that whitespace doesn't matter.

File: basic-javascript/project-1.spec.js [hunk 2]
```diff
@@ -1,3 +1,124 @@
...
+	expect(helpers.getGreeting("Patrick")).toBe("Hello Patrick!");
+	expect(helpers.getGreeting(" Patrick ")).toBe("Hello Patrick!");
+});
+test("should give area of a rectangle", () => {
+	expect(helpers.getRectangleArea(3, 5)).toBe(15);
+	expect(helpers.getRectangleArea("3", 5)).toBe(15);
+});
+test("should give area of a triangle", () => {
+	expect(helpers.getTriangleArea(4, 4)).toBe(8);
+	expect(helpers.getTriangleArea("4", 4)).toBe(8);
```
Comment (bradencodes) [inline basic-javascript/project-1.spec.js:117]: Good test to show that you can use numbers as strings.

File: basic-javascript/project-2.spec.js [hunk 3]
```diff
@@ -1,3 +1,110 @@
...
+		expect(funcs.isPrime(20)).toBe(false);
+	});
+});
+describe("array tests", () => {
+	let arr = [1, 2, 3, 4, 5];
+	let arruno = [1, 2, 3, 4, 5];
+	let arrdos = [1, 2, 3, 4, 5];
+	let arrAddEnd = [1, 2, 3, 4, 5, 6];
+	let arrAddFront = [0, 1, 2, 3, 4, 5];
+	let arrOne = [2, 3, 4, 5, 6];
```
Comment (bradencodes) [inline basic-javascript/project-2.spec.js:90]: You can make the functions immutable so that it doesn't change the array that you pass into it. Then you won't need to create an array for every test.

**Comment by bradencodes** [comment]
## MVP
- [x] Write tests for project-1.js
- [x] Write tests for project-2.js

## Stretch
- [ ] Correct bugs and refactor the code
- [ ] Write tests for arrays.js
- [ ] Write tests for object.js

## What Looks Great
- MVP complete! Nice work
- Frequent and descriptive commits on day 2
- Code is well formatted and easy to read
- Titles for tests are descriptive
- Tests on day 2 are thorough

## What Could Be Improved
- Need more commits on day one
- Tests should check for expected failures and edge cases as well. Don't only test things that should work.

## Rating (1-3)
**2**
